test/kongintegration/containers/kong.go

```diff
@@ -70,9 +71,21 @@ func NewKong(ctx context.Context, t *testing.T, opts ...KongOpt) Kong {
 
 	reqCtx, cancel := context.WithTimeout(ctx, test.RequestTimeout)
 	defer cancel()
-	kongVersion, err := helpers.ValidateMinimalSupportedKongVersion(reqCtx, adminURL, consts.KongTestPassword)
-	require.NoError(t, err)
-	fmt.Printf("INFO: using Kong instance (version: %q) reachable at %s\n", kongVersion, adminURL)
+
+	const (
+		waitTime = time.Minute
+		tickTime = 100 * time.Millisecond
+	)
+	require.Eventually(t, func() bool {
+		kongVersion, err := helpers.ValidateMinimalSupportedKongVersion(reqCtx, adminURL, consts.KongTestPassword)
+		if err != nil {
```

Comment:
Good point, taken this error into account. PTAL

Reply:
I was thinking rather about calling `t.Fatal(...)` when a returned error is `helpers.TooOldKongGatewayError{}`. But now I'm thinking whether is suitable since we spin some stuff...

What happens now when too old version is used? Ir proceeds, what is not desirable.

Reply:
Previous version was equivalent to `t.Fatal()`. Now you got be thinking.

Before it would fail on the spot. Now it continues. If you want to test older image you're welcome to do so.

I believe we should revert to the old version though. If we enforce 3.4+ then I don't se a reason not to do it here